### PR TITLE
Engines Connector

### PR DESCRIPTION
**Pull request by joemcelroy**
This is a fork of the Elasticsearch connector with some key differences:
- Overrides the default Query Processor for Searchkit to use a single `combined_fields` query.
- The connector configuration is `apiKey`, `engineName` and `host`. All three are needed. We do not allow to add custom headers.
- Changed the client meta headers to `engine-connector`

Updated the tests.

### CODE REVIEW

File: packages/search-ui-engines-connector/src/handlers/autocomplete/index.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+  const transporter = new EngineTransporter(host, engineName, apiKey);
+
+  const response = await Searchkit(
+    searchkitConfig,
+    transporter
+  ).executeSuggestions(state.searchTerm);
+
+  const results: AutocompleteResponseState = response.reduce(
+    (sum, suggestion) => {
```
Comment (sloanelybutsurely) [inline packages/search-ui-engines-connector/src/handlers/autocomplete/index.ts:105]: `sum` was a bit confusing to me first. I was expecting numbers. Do you mind using `acc` like you've done [elsewhere](https://github.com/elastic/search-ui/pull/919/files#diff-c60b287c9253b1948b77d2a62817e2112b694ba6f9a6b8f4ef897c26525827c7R13)

File: packages/search-ui-engines-connector/src/handlers/search/Request.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  size: number;
+  sort: string;
+}
+
+export function getFilters(
+  filters: Filter[] = [],
+  baseFilters: Filter[] = []
+): MixedFilter[] {
+  return filters.reduce((acc, f) => {
+    const isBaseFilter = baseFilters.find((bf) => bf === f);
```
Comment (sloanelybutsurely) [inline packages/search-ui-engines-connector/src/handlers/search/Request.ts:19]: nit: i think this could be `baseFilters.includes(f)`

**Review by sloanelybutsurely — APPROVED** [review]

File: packages/search-ui-engines-connector/src/handlers/autocomplete/index.ts (hunk 1 above)
Comment (joemcelroy) [reply packages/search-ui-engines-connector/src/handlers/autocomplete/index.ts]: Good suggestion. thanks!

File: packages/search-ui-engines-connector/src/handlers/search/Request.ts (hunk 2 above)
Comment (joemcelroy) [reply packages/search-ui-engines-connector/src/handlers/search/Request.ts]: agreed. Thanks!

**Comment by joemcelroy** [comment]
@sloanelybutsurely i've also applied the changes to elasticsearch-connector too. Thank you for reviewing!

**Review by sloanelybutsurely — APPROVED** [review]

**Pull request merged**